### PR TITLE
Don't Use Special IPv6 Addresses

### LINKED ISSUE #790: Python Cert Tests Shouldn't Use Special IPv6 Addresses
Currently, the Python cert tests use 2001::/64 and 2002:/64, both of which fall into special IPv6 address groups:

https://en.wikipedia.org/wiki/IPv6_address#Unicast_addresses

### PR DESCRIPTION
**Pull request by nibanks**
This fixes #790.

### CODE REVIEW

**Comment by darconeous** [comment]
I'm not sure `3000::/3` is a better long term choice. Isn't that unallocated?

Is there a range explicitly designated for testing that we could use instead?

**Comment by nibanks** [comment]
From Wikipedia, the closest I could find:

[**Documentation**
](https://en.wikipedia.org/wiki/IPv6_address#Documentation)

- **2001:db8::/32** - This prefix is used in documentation. The addresses should be used anywhere an example IPv6 address is given or model networking scenarios are described (corresponding to 192.0.2.0/24, 198.51.100.0/24, and 203.0.113.0/24 in IPv4.)

**Comment by LuDuda** [comment]
I  agree with @darconeous for this purpose i would also use 2001:db8::/32 prefix.

Look here:
https://tools.ietf.org/html/rfc3849
http://www.iana.org/assignments/ipv6-address-space/ipv6-address-space.xhtml

**Comment by darconeous** [comment]
@LuDuda: The only caveat is that this is a test harness, not documentation.

I'm not sure if there is a good answer for what prefix to use, only which ones might be undesirable.

We could always use a ULA.

**Comment by nibanks** [comment]
According to the [iana link](http://www.iana.org/assignments/ipv6-address-space/ipv6-address-space.xhtml) @LuDuda shared, nothing in the `3000::/3` range is reserved. I don't think you can reserve things after the fact, so it shouldn't become reserved at a later date.

**Comment by LuDuda** [comment]
Maybe i'm not correct, however i see that you used for example 3003::/64 prefix, which is inside 2000::/3 space and thus it may be allocated by someone as a global prefix.

**Comment by jwhui** [comment]
IANA is the authority for [IPv6 Address Space allocations](http://www.iana.org/assignments/ipv6-address-space/ipv6-address-space.xhtml).  As instructed by [RFC 6890](https://tools.ietf.org/html/rfc6890), the `2001:2::/48` prefix was reserved specifically for benchmarking and testing purposes.  Maybe we should be using that?

**Comment by LuDuda** [comment]
I think it may be a good approach, also i found information about it here (specifically in section 8):
https://tools.ietf.org/html/rfc5180

**Comment by nibanks** [comment]
That sounds fine with me. So what do you want to explicitly change the following to? I will update my PR accordingly.

`2001::/64` -> `?`
`2002::/64` -> `?`
`2003::/64` -> `?`
`2004::/64` -> `?`

**Comment by jwhui** [comment]
Yes, [RFC 5180](https://tools.ietf.org/html/rfc5180) provides additional background.  However, note that Section 8 actually specifies a prefix different from IANA.  Ultimately, IANA is the authority for such allocations.

`2001::/64` -> `2001:2:0:1::/64`
`2002::/64` -> `2001:2:0:2::/64`
`2003::/64` -> `2001:2:0:3::/64`
`2004::/64` -> `2001:2:0:4::/64`

I'd also like to point out that the existing prefixes used by the python test scripts are compliant with the Thread Certification Test Specification.  We'll need to work with the Thread Group to update the test spec.

**Comment by nibanks** [comment]
A question for the change:

Should `'2001' in addr[0:4]` become `'2001:2:0:1' in addr[0:10]`?

**Comment by jwhui** [comment]
Yes, I believe we need to update the string match checks as well.

File: tests/scripts/thread-cert/Cert_5_6_04_NetworkDataRegisterAfterAttachRouter.py [hunk 1]
```diff
@@ -89,24 +89,24 @@ def test(self):
...
+        self.nodes[ROUTER].add_prefix('2001:2:0:1::/64', 'paros')
+        self.nodes[ROUTER].add_prefix('2001:2:0:2::/64', 'paro')
         self.nodes[ROUTER].register_netdata()
 
         time.sleep(10)
 
         addrs = self.nodes[ED1].get_addrs()
-        self.assertTrue(any('2001' in addr[0:4] for addr in addrs))
-        self.assertTrue(any('2002' in addr[0:4] for addr in addrs))
+        self.assertTrue(any('2001:2:0:1' in addr[1:10] for addr in addrs))
```
Comment (nibanks) [inline tests/scripts/thread-cert/Cert_5_6_04_NetworkDataRegisterAfterAttachRouter.py]: I don't know how, but somehow 1:10 snuck in here...

**Pull request merged**